pilot/pkg/xds/ads_common.go

```diff
@@ -23,7 +23,6 @@ import (
 // configKindAffectedProxyTypes contains known config types which will affect certain node types.
 var configKindAffectedProxyTypes = map[resource.GroupVersionKind][]model.NodeType{
 	gvk.Gateway: {model.Router},
-	gvk.Sidecar: {model.SidecarProxy},
```

Comment:
Any reason for this change? The intent is to not have Sidecar impact Gateway

Reply:
It causes `ConfigAffectsProxy` to return true without checking `DependsOnConfig`, I'll add a testcase that this change will break and fix the logic.